src/NServiceBus.Core/IBusExtensions.cs

```diff
@@ -81,7 +81,7 @@ public static void Subscribe(this IBus bus, Type messageType)
             Guard.AgainstNull("bus", bus);
             Guard.AgainstNull("messageType", messageType);
 
-            bus.Subscribe(messageType, new SubscribeOptions());
+            bus.SubscribeAsync(messageType, new SubscribeOptions());
```

Comment:
We need to fix this. The synchronous version should be obsoleted and only the async version should be exposed